swift-paperless/Utilities/PKCS12.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  PKCS12.swift
+//  swift-paperless
+//
+//  Created by Nils Witt on 24.06.24.
+//
+
+import Foundation
+
+enum PKCS12Error: Error {
+    case wrongPassword
+    case other
+}
+
+class PKCS12 {
+    let identity: SecIdentity?
+
+    public init(pkcs12Data: Data, password: String) throws {
+        let importPasswordOption: NSDictionary = [kSecImportExportPassphrase as NSString: password]
+        var items: CFArray?
+        let secError: OSStatus = SecPKCS12Import(pkcs12Data as NSData, importPasswordOption, &items)
+        guard secError == errSecSuccess else {
+            if secError == errSecAuthFailed {
+                throw PKCS12Error.wrongPassword
+            }
+            throw PKCS12Error.other
+        }
+        guard let theItemsCFArray = items else { fatalError() }
```

Comment:
Can this ever happen? Consider adding a case to `PKCS12Error for this.

Reply:
This and the one below should never fail as the SecPKCS12Import function should not resolve to success when there were a problem and they should always be set that way. 
But for better handling I´ll add better error handling